editor.planx.uk/src/pages/FlowEditor/components/Flow/FeedbackPage.tsx

```diff
@@ -46,20 +83,188 @@ const getDetailedFeedback = async (feedbackId: number) => {
+  displayFeedbackItems: string[];
+}
+
+const CollapsibleRow: React.FC<CollapsibleRowProps> = (item) => {
+  const [open, setOpen] = useState<boolean>(false);
+  const [detailedFeedback, setDetailedFeedback] = useState<
+    Record<string, any> | undefined
+  >(undefined);
+  const { icon, title } = feedbackTypeIcon(item.type);
+
+  const toggleDetailedFeedback = async () => {
+    setOpen(!open);
+    if (!open && !detailedFeedback) {
+      const fetchedData = await getDetailedFeedback(item.id);
+      setDetailedFeedback(fetchedData);
+    }
+  };
+
+  const commentSummary = item.userComment
+    ? item.userComment.length > 50
+      ? `${item.userComment.slice(0, 50)}...`
+      : item.userComment
+    : "No comment";
+
+  const filteredFeedbackItems = (() => {
+    switch (item.type) {
+      case "issue":
+        return ["userContext", ...item.displayFeedbackItems];
+      case "helpful":
+      case "unhelpful":
+        return ["helpText", ...item.displayFeedbackItems];
+      default:
+        return item.displayFeedbackItems;
+    }
+  })();
+
   return (
-    <Box sx={{ fontSize: 12, overflowY: "auto" }}>
-      {feedback.map((item) => (
-        <React.Fragment key={item.id}>
-          <Box component="pre">{JSON.stringify(item, null, 4)}</Box>
-          <Button onClick={() => getDetailedFeedback(item.id)}>
-            Log out detailed info
-          </Button>
-        </React.Fragment>
-      ))}
-    </Box>
+    <React.Fragment>
+      <TableRow key={item.id}>
+        <TableCell>
+          <Box sx={{ display: "flex", alignItems: "center", gap: 1 }}>
+            {icon} {title}
+          </Box>
+        </TableCell>
+        <TableCell>
+          {format(new Date(item.createdAt), "dd/MM/yy hh:mm:ss")}
+        </TableCell>
+        <TableCell>{commentSummary}</TableCell>
+        <TableCell sx={{ textAlign: "right" }}>
+          <IconButton
+            aria-label="expand row"
+            size="small"
+            onClick={toggleDetailedFeedback}
+          >
+            {open ? <KeyboardArrowUp /> : <KeyboardArrowDown />}
+          </IconButton>
+        </TableCell>
+      </TableRow>
+      <TableRow sx={{ background: (theme) => theme.palette.background.paper }}>
+        <TableCell sx={{ padding: 0, border: "none" }} colSpan={4}>
+          <Collapse
+            in={open}
+            timeout="auto"
+            unmountOnExit
+            sx={{
+              borderBottom: (theme) =>
+                `1px solid ${theme.palette.border.light}`,
+              padding: (theme) => theme.spacing(0, 1.5),
+            }}
+          >
+            <DetailedFeedback>
+              <SummaryListTable sx={{ margin: "0" }}>
+                {detailedFeedback &&
+                  filteredFeedbackItems.map(
+                    (key, index) =>
+                      detailedFeedback[key] !== undefined && (
+                        <React.Fragment key={index}>
+                          <Box component="dt">{key}</Box>
```

Comment:
To display a more human-friendly label here we could create a map / object to handle this - 

```ts
const labelMap = {
    "userComment": "User comment",
    "address": "Address",
    "projectType": "Project type",
    ...etc
}
```

Then we could reference it here like this - 

```ts
<Box component="dt">{labelMap[key]}</Box>
```

Reply:
Thanks Daf, I've added this